File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int s = 0;
        int f = 1;


        int [] ans = new int[2];
        while(s<=nums.Length-2){
            if(nums[s] + nums[f] == target)
            {
                ans[0] = s;
                ans[1] = f;
            }
            if(f== nums.Length-1){
                s++;
                f = s+1;
            }
            else{
            f++;
            }
        }

        return ans;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        vector<int> ans;
        vector<int> array;
        
         int n = nums.size();
         int a[n];
        for(int i=0;i<n;i++){
                a[i] = nums[i];
        }
        
      
        sort(nums.begin(),nums.end());
        
        int start = 0;
        int end = n-1;
 
        while(start<end){
            if(nums[start]+nums[end]==target){
                 array.push_back(nums[start]);
                array.push_back(nums[end]);
                break;
            }
            else if(nums[start]+nums[end]>target){
                end--;
            }
            else{
                    start++;
            }
        }
      
        for(int i=0 ;i<n;i++){
            for(int j=0;j<array.size();j++){
                if(a[i] == array[j]){
                   
                    ans.push_back(i);
                    break;

                }
            }
        }



        //  for(int i=0 ;i<n;i++){
        //     for(int j=0;j<n;j++){
        //             if(nums[i]+nums[j] == target){
        //                 ans.push_back(i);
        //                 ans.push_back(j);
        //                 break;
        //             }
        //     }
        // }



        return ans;
    }
};